functional_tests/steps/bundles.py

```diff
@@ -121,3 +126,166 @@ def the_user_can_see_the_bundle_details_with_creator(context: Context) -> None:
 def bundle_inspect_show(context: Context) -> None:
     expect(context.page.get_by_text("Created by")).to_be_visible()
     expect(context.page.get_by_text(context.bundle_creator.get_username())).not_to_be_visible()
+
+
+# Bundles UI Data Setup
+@given("there are {no_bundles} bundles")
+def multiple_bundles_create(context: Context, no_bundles: str) -> None:
+    context.bundles = []
+    if no_bundles.isdigit():
+        for __ in range(int(no_bundles)):
+            context.bundles.append(BundleFactory(approved=False, created_by=context.user_data["user"]))
+
+
+@given("there are {no_preview_teams} Preview teams")
+def multiple_preview_teams_create(context: Context, no_preview_teams: str) -> None:
+    context.teams = []
+    if no_preview_teams.isdigit():
+        for preview_team_index in range(int(no_preview_teams)):
+            context.teams.append(
+                Team.objects.create(
+                    identifier="preview-team-" + str(preview_team_index), name="Preview_Team_" + str(preview_team_index)
+                )
+            )
+
+
+@given("there are {no_release_calendar} release calendar pages")
+def multiple_release_calendar(context: Context, no_release_calendar: str) -> None:
+    context.release_calendar_pages = []
+    if no_release_calendar.isdigit():
+        for release_calendar_index in range(int(no_release_calendar)):
+            nowish = timezone.now() + timedelta(minutes=5 * (release_calendar_index + 1))
+            context.release_calendar_pages.append(
+                ReleaseCalendarPageFactory(
+                    release_date=nowish,
+                    title="Release Calendar Page" + str(release_calendar_index),
+                    status=ReleaseStatus.CONFIRMED,
+                )
+            )
+
+
+@given("there are {no_statistical_analysis} Statistical Analysis pages")
+def multiple_statistical_analysis(context: Context, no_statistical_analysis: str) -> None:
+    context.statistical_article_pages = []
+    if no_statistical_analysis.isdigit():
+        for statistical_analysis_index in range(int(no_statistical_analysis)):
+            context.statistical_article_pages.append(
+                StatisticalArticlePageFactory(
+                    parent=ArticleSeriesPageFactory(title="PSF" + str(statistical_analysis_index))
+                )
+            )
+
+
+@given("there is a {user_role} user")
+def creat_user_by_role(context: Context, user_role: str) -> None:
+    context.user_data = create_user(user_role)
+
+
+@given("the {user_role} is a member of the Preview teams")
```

Comment:
You've defined a parameter here `{user_role}` but it doesn't seem to be used by the function itself, so I think it can be dropped. If we keep it as "the user is a member of the Preview teams", it should be clear from the context that it's the user we created.